sklego/feature_selection/mrmr.py

```diff
@@ -0,0 +1,216 @@
+
+
+class MaximumRelevanceMinimumRedundancy(SelectorMixin, BaseEstimator):
+    """Maximum Relevance Minimum Redundancy (MRMR) is an iterative feature selection method commonly used in data
+    science to select a subset of features from a larger feature set. The goal of MRMR is to choose features that
+    have high relevance to the target variable while minimizing redundancy among the already selected features.
+
+    How MRMR works:
+
+    1. Compute the relevance of each feature to the target variable: The relevance of a feature is typically
+    measured using a metric such as mutual information, correlation coefficient, or another appropriate measure of
+    dependence between the feature and the target variable.
+
+    2. Compute the redundancy between each pair of features: Redundancy is the degree of similarity or overlap between
+    features. It can be measured using metrics such as mutual information, correlation coefficient, or other similarity
+    measures.
+
+    3. Select features based on the maximum relevance and minimum redundancy criteria: MRMR aims to maximize the
+    relevance of selected features to the target variable while minimizing redundancy among them.
+
+    4. Construct the final subset of features: MRMR iteratively adds features to the selected subset until a predefined
+    number of features is reached.
+
+    Parameters
+    ----------
+    k : int
+        Number of feature the model should use.
+    relevance_func : str | Callable,, default= "f"(f_classif or  f_regression from sklearn.feature_selection)
+        [f_classif](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.f_classif.html)
+        [f_regression](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.f_regression.html)
+        The relevance function to use.
+    redundancy_func : str | Callable, default="p" (Pearson correlation)
+        The redundancy function to use.
+    kind : Literal["auto", "classficiation", "regression"], default="auto".
+        'classification' or 'regression' or 'auto' if auto the model
+        will try to infer the type of problem looking at the y data type, by default "auto".
+
+    !! warning:
+        If a custom relevance_func is provided it must have this signature:
+        Callable[[np.ndarray, np.ndarray], np.ndarray]
+        It should accept X, y as arguments and it should compute the score for each feature of X
+        and return an array of shape (n_features_in_,).
+    !! warning:
+        If a custom redundancy_func is provided it must have the same signature as the method _redundancy_pearson
+
+    Attributes
+    ----------
+    _y_dtype : data type of y
+    selected_features_ : array-like of shape (k,)
+        Indexes of the selected features.
+    scores_ : array-like of shape (k,)
+        Scores of the selected features.
+
+    Examples
+    --------
+    ```py
+    from sklego.feature_selection.mrmr import MaximumRelevanceMinimumRedundancy
```

Comment:
I would prefer to have a similar logic as in the other submodules. Namely:

- In `sklego.feature_selection.__init__.py` add the import for `MaximumRelevanceMinimumRedundancy`
- Then one is able to import from feature selection module
     ```
     from sklego.feature_selection import MaximumRelevanceMinimumRedundancy
     ```